demo-simple/src/main/java/com/novoda/downloadmanager/demo/DownloadBatchSizeRequirementRule.java

```diff
@@ -5,6 +5,8 @@
 
 public class DownloadBatchSizeRequirementRule implements DownloadBatchRequirementRule {
 
+    public final static int CODE = DownloadBatchRequirementRule.class.hashCode();
```

Comment:
For instance. This is just the demo app.

Reply:
I think that both name and value of this field are misleading. I'd change to to something like:
`public final static int ERROR_CODE_DOWNLOAD_LIMIT_REACHED = 1;`
There's also a problem that with integers it's hard to make sure that the code in different rules are unique, so maybe a string value would be better?
`public final static int ERROR_CODE_DOWNLOAD_LIMIT_REACHED = "ERROR_CODE_DOWNLOAD_LIMIT_REACHED";`

Reply:
Well, technically someone could use identical string values for different codes just the same. The `class.hashCode()` trick sort of deals with the problem of uniqueness. But maybe it's indeed easier and more human-readable when it's a String. I'll do so.

Reply:
One problem with that suggested change is that it leads to non-unique `DownloadError` constructors.

Currently `DownloadError` takes a `Type` - and, optionally, a `String` message.

The fact that both would be of `String` type means you can no longer have constructors emulating optional parameters - as it creates ambiguity. And from developer's perspective, it's now possible to confuse a message with a textual code, and vice versa, since the type system won't help you to differentiate them. 

Of course it could be mitigated by adding factory methods, but again this increases the calibre of changes we'd make to the library.

Or maybe the code should be a custom type (value wrapper) rather than a primitive type? What do you reckon

Reply:
OK, so maybe we keep int value and just change the name

Reply:
All right